src/extension/background-script/actions/lnurl/authOrPrompt.ts

```diff
@@ -33,12 +34,25 @@ async function authOrPrompt(
   // we have the check the unlock status manually. The account can still be locked
   // If it is locked we must show a prompt to unlock
   const isUnlocked = await state.getState().isUnlocked();
+  const account = await state.getState().getAccount();
+  const isAlbyOAuthConnector = isAlbyOAuthAccount(account?.connector);
 
   // check if there is a publisher and lnurlAuth is enabled,
   // otherwise we we prompt the user
+
   if (isUnlocked && allowance && allowance.enabled && allowance.lnurlAuth) {
-    return await authFunction({ lnurlDetails, origin: message.origin });
+    if (!isAlbyOAuthConnector) {
+      return await authFunction({ lnurlDetails, origin: message.origin });
+    } else if (account?.mnemonic) {
```

Comment:
could you could replace `else if` with 
`if (!isAlbyOAuthConnector || account?.mnemonic) {` ?

Reply:
the logic needs to handle below cases, how this suggestion handles that?

1. if allowance is enabled and it's not oauth connector. just return directly as no keys required
2. if allowance is returned and if it's alby connector we need to check keys. if keys present return dirrectly
3. if allowance is enabled and keys are not present. open auth prompt
4. if allownce is not enabled return auth prompt

Reply:
sry. i missread that. corrected